lib/express.js

```diff
@@ -32,16 +57,22 @@ exports = module.exports = createApplication;
  * @api public
  */
 
-function createApplication() {
+function createApplication(opts) {
+  opts = opts || {}
+  opts.reqProto = opts.reqProto || http.IncomingMessage.prototype
+  opts.resProto = opts.resProto || http.ServerResponse.prototype
+  opts.createServer = opts.createServer || http.createServer
+
   var app = function(req, res, next) {
     app.handle(req, res, next);
   };
 
   mixin(app, EventEmitter.prototype, false);
   mixin(app, proto, false);
 
-  app.request = { __proto__: req, app: app };
-  app.response = { __proto__: res, app: app };
+  app.createServer = opts.createServer
+  app.request = setPrototypeOf({ app: app }, req(opts.reqProto))
+  app.response = setPrototypeOf({ app: app }, res(opts.resProto))
```

Comment:
I believe these two lines could just be `Object.create(proto, descriptorsWithJustApp)`, right?
```js
app.response = Object.create(res(opts.resProto), {
  app: { configurable: true, enumerable: true, writable: true, value: app }
})
```